src/hidra/receiver/plugins/asapo_producer.py

```diff
@@ -281,18 +293,18 @@ def _callback(self, header, err):
             logger.error("Could not sent: %s, %s", header, err)
 
     def _parse_file_name(self, path):
-        matched = parse_file_path(self.file_regex, path)
         try:
-            data_source = get_entry(matched, "data_source")
-        except utils.UsageError:
+            matched = parse_file_path(self.file_regex, path)
             if self.default_data_source is not None:
                 data_source = self.default_data_source
             else:
-                raise
+                data_source = get_entry(matched, "data_source")
 
-        stream = get_entry(matched, "scan_id")
-        file_idx = int(get_entry(matched, "file_idx_in_scan"))
-        return data_source, stream, file_idx
+            stream = get_entry(matched, "scan_id")
+            file_idx = int(get_entry(matched, "file_idx_in_scan"))
+            return data_source, stream, file_idx
+        except Exception as e:
```

Comment:
Which errors does this catch and why?

For me it looks like the only possible error is `ValueError` from the conversion to `int`. I would implement and document the following semantics for `_parse_file_name`:

0. It is assumed that the regex is always a valid, compiled regex object and `path` is a `str`
1. If the filename doesn't match the regex, return `None`
2. If the filename matches the regex, but still cannot be parsed, raise `ValueError`
3. Otherwise, return `data_source: str`, `stream: str`, `file_idx: int`

With this API, `send_message` can handle cases 1. and 2. differently. 1. can be a simple debug log as this is expected behavior, but 2. should be a warning.